### ISSUE TITLE
sending mail error

### ISSUE THREAD

**Issue by idiengaid** [vahidhedayati/testmlist#1]
Hello me again
i was on the mailinglist plugin installed, configured work fine but in my app when i try to send a group mail a got an exception on console for sending mail
2017-02-17 21:30:04,530 [quartzScheduler_Worker-2] ERROR core.MailingListEmailService  - Problem sending email message has no content, use text(), html() or body() methods to set content
Message: message has no content, use text(), html() or body() methods to set content
   Line | Method
->> 433 | finishMessage  in grails.plugin.mail.MailMessageBuilder
....

and also on screen on view redirected
![image](https://cloud.githubusercontent.com/assets/19474979/23084070/9c696324-f558-11e6-8c5f-64ff28a78c85.png)

same conf
win 7
ggts
grails 244
hope will be helped tks

**Comment by idiengaid** [comment]
Hello for error on the page it's solved on th view _list1-top.gsp on mailling list plugin on line 61 an "i" were missing i don't know how on instance like that

![image](https://cloud.githubusercontent.com/assets/19474979/23084583/f7fab3ee-f55a-11e6-9308-b7d91d480950.png)

but now i correct and fine.
now let me look mail probleme

**Comment by vahidhedayati** [comment]
Thanks :+1:  I have now updated it all and released a new version (on grails 2 just downloading the libraries so will be updated very shortly)

Thanks again for your time and effort.

**Comment by idiengaid** [comment]
for the community

**Comment by idiengaid** [comment]
hello this is th error
![image](https://cloud.githubusercontent.com/assets/19474979/23086659/3dd394c0-f566-11e6-90ed-fe4ea548d5cc.png)
my config is
grails {
	mail {
	  host = "smtp.gmail.com"
	  port = 465
	  username = "xxxxxxx@gmail.com"
	  password = "xxxxxxxx"
	  props = [
			   "mail.smtp.auth"                  : "true",
			   "mail.smtp.socketFactory.port"    : "465",
			   "mail.smtp.socketFactory.class":"javax.net.ssl.SSLSocketFactory",
			   "mail.smtp.socketFactory.fallback":"false"]
	  }
}

and it's work if i user sendMail
sendMail {
    to params.adress
    subject params.title
    body params.body
}
regards

**Comment by vahidhedayati** [comment]
[Line 182 ](https://github.com/vahidhedayati/mailinglist/blob/grails2/grails-app/services/grails/plugin/mailinglist/core/MailingListEmailService.groovy#L182)is this under grails 2 plugin 
					if (paramsMap.attachments) {


error is around message being blank you should see a html editable area to fill in the message ensure you have some text in there...

Also obviously if you are trying attachments try without to see if it works to narrow down if that is issue

**Comment by idiengaid** [comment]
Ok i've fine it.
it's look like html is not working and body parmeter is need i've changed on line 198 for bulk
![image](https://cloud.githubusercontent.com/assets/19474979/23087352/83baab5e-f56b-11e6-9c96-77b7df55d990.png)
also on line 251 for individual
![image](https://cloud.githubusercontent.com/assets/19474979/23087370/a8e57b84-f56b-11e6-8238-bd7165b67f32.png)
and also on line 299 for else
![image](https://cloud.githubusercontent.com/assets/19474979/23087383/ce117020-f56b-11e6-8452-20aec8ecd88c.png)

the only matter is that message is comming with http balise like that
![image](https://cloud.githubusercontent.com/assets/19474979/23087409/0f20abee-f56c-11e6-91b4-95da20e19155.png)

have a look on it please
PS 182 under grails 2 is mailService.sendMail closure
![image](https://cloud.githubusercontent.com/assets/19474979/23087460/7f351456-f56c-11e6-85fb-05c15aae4b37.png)

**Comment by vahidhedayati** [comment]
Thanks for the reporting the issue.

Basically the issue was from this:

> we have 1 class java.lang.String

```
println "we have $template ${template.getClass()}"
if(template){
 html grailsApplication.mainContext.groovyPageRenderer.render(template: template, model: templateModel)
} else {
 html message
}

```
For some reason it was returning a 1 and and picked up by the if template segment and then failing to render a template for a template that was called 1.

The template stuff was added by another member and in all honesty i don't really use this plugin, I have tweaked the ui a little since it was not working too well and if I had the time i would rewrite it since it is rather messy.


Anyhow the fix is this :  


```
if(template && templateModel){
  .. do what it does
}else {
  html message
}

```

Since although you included the message as body you found it converted it into actual html - this is because it has to be as it is currently html message since it is a html message.

Anyhow I didn't rush it last night since I was tired and having had a look at your suggestion that obviously would then break the if statements doing this logic below.

I will fix it soon in the different bits and roll out an update I need to update version 3 of the plugin too.

As i say i wish I had the time to rewrite it a lot of the mess could be done in so much less lines

**Comment by idiengaid** [comment]
Hello very well it's working i've seen the error about template. But don't worry if u need some one to help i'm there. See you soon every think fixed

**Comment by vahidhedayati** [comment]
hope it all works well. It is entirely up to you there is no pressure. Each time i return to this plugin I see all the params in the controllers the messy gsp bad labelling incorrect front end validation the modal box stuff can just be replaced with jquery dialog box doing local stuff rather rather than bootstrap modal box. 

Ideally if i had the time in the world I would like to introduce [backend bean validation](https://github.com/vahidhedayati/grails-queuemail-plugin/blob/master/grails-app/controllers/org/grails/plugin/queuemail/QueueMailController.groovy) like in this plugin then validation will just happen automagically rather than all those if statements hard to follow around. The rest of it needs a fine combe walk through 

Unsure of your level of comfort, someone had written a documentation step by step guide for this plugin for my last place but kind of went into their business world so hasn't been included although it takes 5 minutes to get your head around it

**Comment by vahidhedayati** [comment]
ps do let me know before starting anything as a starter it is either worth updating to a later version of grails or repackaging a given plugin to new package name of `org.grails.plugins` from `grails.plugins` since this was the nightmare I faced whilst doing one of the plugins it turned out to be something in earlier 2.4 grails and reloading was not enabled for anything that started with package name of grails.plugins which means upon any change you need to restart the app to test changes in short just a total night mare as it is at the moment.
